patches/server/1061-add-disguise-api.patch

```diff
@@ -0,0 +1,440 @@
+         Set<AttributeInstance> set = this.getAttributes().getAttributesToUpdate();
++        if (io.papermc.paper.disguise.DisguiseUtil.shouldSkipAttributeSending(this)) return; // Paper - disguise api
+         Iterator iterator = set.iterator();
+ 
+         while (iterator.hasNext()) {
+@@ -2577,6 +2578,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
+             this.swinging = true;
+             this.swingingArm = hand;
+             if (this.level() instanceof ServerLevel) {
++                if(!io.papermc.paper.disguise.DisguiseUtil.canSendAnimation(this)) return; // Paper - disguise api
+                 ClientboundAnimatePacket packetplayoutanimation = new ClientboundAnimatePacket(this, hand == InteractionHand.MAIN_HAND ? 0 : 3);
+                 ServerChunkCache chunkproviderserver = ((ServerLevel) this.level()).getChunkSource();
+ 
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index b185a9b474121b8a4067816b3c3c39270c2a202e..4cead9dd4cf35ce7076f4d33dd9181cedcb5e563 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -383,6 +383,12 @@ public final class CraftServer implements Server {
+         return ca.spottedleaf.moonrise.common.util.TickThread.isTickThreadFor(((org.bukkit.craftbukkit.entity.CraftEntity) entity).getHandleRaw());
+     }
+     // Paper end - Folia reagion threading API
++    // Paper start - add disguise api
++    @Override
++    public com.destroystokyo.paper.SkinParts.@org.jetbrains.annotations.NotNull Builder newSkinPartsBuilder() {
++        return com.destroystokyo.paper.PaperSkinParts.builder();
++    }
++    // Paper end - add disguise api
+ 
+     static {
+         ConfigurationSerialization.registerClass(CraftOfflinePlayer.class);
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+index 179886dcbda29c5cdb7dbd43e44951ae38d9df96..cabc324db333b4298449098d7e7422166831b9c5 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+@@ -1311,4 +1311,18 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
+         }
+     }
+     // Paper end - broadcast hurt animation
++    // Paper start - disguise api
++    private io.papermc.paper.disguise.DisguiseData disguiseData = io.papermc.paper.disguise.DisguiseData.original();
++    @Override
++    public @org.jetbrains.annotations.NotNull io.papermc.paper.disguise.DisguiseData getDisguiseData() {
++        return disguiseData;
++    }
++
++    @Override
++    public void setDisguiseData(@org.jetbrains.annotations.NotNull io.papermc.paper.disguise.DisguiseData disguiseData) {
++        getHandle().moonrise$getTrackedEntity().moonrise$clearPlayers();
```

Comment:
The following lines ensure that a force refresh is triggered for all players that can see the entity.